src/main/java/org/fenixedu/academic/ui/struts/action/academicAdministration/executionCourseManagement/ExecutionCourseBean.java

```diff
@@ -122,7 +149,62 @@ private DegreeCurricularPlan getDegreeCurricularPlan() {
         return getExecutionDegree().getDegreeCurricularPlan();
     }
 
-    public ExecutionCourseBean() {
+    public String getSourcePresentationName() {
+        String result = StringUtils.EMPTY;
+
+        if (getSourceExecutionCourse() != null) {
+            result += getSourceExecutionCourse().getNameI18N().getContent();
+
+            final Set<DegreeCurricularPlan> plans;
+            if (getDegree() != null) {
+                plans =
+                        Sets.intersection(getDegree().getDegreeCurricularPlansSet(),
+                                Sets.newHashSet(getSourceExecutionCourse().getAssociatedDegreeCurricularPlans()));
+            } else {
+                plans = Sets.newHashSet(getSourceExecutionCourse().getAssociatedDegreeCurricularPlans());
+            }
+
+            result += getDegreeCurricularPlansPresentationString(plans);
+        }
+
+        return result;
+    }
+
+    public String getDestinationPresentationName() {
+        String result = StringUtils.EMPTY;
+
+        if (getDestinationExecutionCourse() != null) {
+            result += getDestinationExecutionCourse().getNameI18N().getContent();
+
+            final Set<DegreeCurricularPlan> plans;
+            if (getDegree() != null) {
+                plans =
+                        Sets.intersection(getDegree().getDegreeCurricularPlansSet(),
+                                Sets.newHashSet(getDestinationExecutionCourse().getAssociatedDegreeCurricularPlans()));
+            } else {
+                plans = Sets.newHashSet(getDestinationExecutionCourse().getAssociatedDegreeCurricularPlans());
+            }
+
+            result += getDegreeCurricularPlansPresentationString(plans);
+        }
+
+        return result;
+    }
+
+    static private String getDegreeCurricularPlansPresentationString(final Set<DegreeCurricularPlan> input) {
+        String result = " [ ";
+
+        for (Iterator<DegreeCurricularPlan> iterator = input.iterator(); iterator.hasNext();) {
+            final DegreeCurricularPlan iter = iterator.next();
+            result += iter.getName();
```

Comment:
This should use a `StringBuilder` (the same goes for all the other usages of `+=` on Strings for this PR).